pcsx2/gui/AppUserMode.cpp

```diff
@@ -15,296 +15,332 @@
+			std::ostringstream os;
+			os << stream;
+			data = os.str();
+			return true;
+		}
+		catch (const std::exception& e)
+		{
+			DevCon.Warning("ERROR: ", e.what());
+			data = "";
+			return false;
+		}
+	}
+	else
+	{
+		return false;
+	}
+}
 
-		FirstTimeWizard wiz( NULL );
-		if( wiz.RunWizard( wiz.GetFirstPage() ) ) break;
-		if (wiz.GetReturnCode() != pxID_RestartWizard)
-			throw Exception::StartupAborted( L"User canceled FirstTime Wizard." );
+YAML::Node Pcsx2App::Save(fs::path fileName)
+{
+	if (!folderUtils.DoesExist(fileName.parent_path().make_preferred()))
+	{
+		folderUtils.CreateFolder(fileName.parent_path().make_preferred());
+	}
 
-		Console.WriteLn( Color_StrongBlack, "Restarting First Time Wizard!" );
+	if (!stream)
+	{
+		try
+		{
+			stream = YAML::Load(fileName.make_preferred());
+		}
+		catch (const std::exception& e)
+		{
+			DevCon.Warning("ERROR: ", e.what());
+		}
 	}
+	std::ofstream fout(fileName.make_preferred());
+	fout << stream;
+
+	return stream;
 }
 
-wxConfigBase* Pcsx2App::OpenInstallSettingsFile()
+bool Pcsx2App::OpenInstallSettingsFile()
 {
 	// Implementation Notes:
 	//
-	// As of 0.9.8 and beyond, PCSX2's versioning should be strong enough to base ini and
+	// As of 0.9.8 and beyond, PCSX2's versioning should be strong enough to base yaml and
 	// plugin compatibility on version information alone.  This in turn allows us to ditch
-	// the old system (CWD-based ini file mess) in favor of a system that simply stores
+	// the old system (CWD-based yaml file mess) in favor of a system that simply stores
 	// most core application-level settings in the registry.
 
-	std::unique_ptr<wxConfigBase> conf_install;
+	InstallationMode = InstallationModeType::InstallMode_Registered;
+	fs::path usrlocaldir = PathDefs::GetUserLocalDataDir();
 
-#ifdef __WXMSW__
-	conf_install = std::unique_ptr<wxConfigBase>(new wxRegConfig());
-#else
-	// FIXME!!  Linux / Mac
-	// Where the heck should this information be stored?
-
-	wxDirName usrlocaldir( PathDefs::GetUserLocalDataDir() );
-	//wxDirName usrlocaldir( wxStandardPaths::Get().GetDataDir() );
-	if( !usrlocaldir.Exists() )
+	InstallFolder = (wxFileName(wxStandardPaths::Get().GetExecutablePath())).GetPath().ToStdString();
+
+	std::string usermodeFile = (GetAppName().ToStdString() + "-reg.yaml");
+	usermodePath = Path::Combine(usrlocaldir, usermodeFile); 
+
+	if (!folderUtils.DoesExist(usermodePath))
 	{
-		Console.WriteLn( L"Creating UserLocalData folder: " + usrlocaldir.ToString() );
-		usrlocaldir.Mkdir();
+		CustomDocumentsFolder = PathDefs::AppRoot();
+
+		// Install Mode, genereate a new stream and manually write data to new file if it doesn't.
+		stream["DocumentsFolderMode"] = (int)DocsFolderMode;
+		stream["CustomDocumentsFolder"] = CustomDocumentsFolder.string();
+		stream["UseDefaultSettingsFolder"] = UseDefaultSettingsFolder;
+		stream["SettingsFolder"] = SettingsFolder.string();
+		stream["Install_Dir"] = InstallFolder;
+		stream["RunWizard"] = false;
```

Comment:
What does this replace?  I couldn't find these strings in the replaced code.  Is this an added feature?

Reply:
> 
> 
> What does this replace? I couldn't find these strings in the replaced code. Is this an added feature?

These replace the registry keys that used to exist under HKCU/PSCX2 on Windows. The new implementation has the same strings but now resides in AppData/Local/PCSX2/PCSX2-reg.yaml

Reply:
Though the above is added. I manually reconstructed the data that we used to pass through WX